### PR TITLE
use DMA channels

### PR DESCRIPTION
**Pull request by burrbull**
Requires https://github.com/esp-rs/esp-pacs/pull/203

### CODE REVIEW

**Comment by burrbull** [comment]
@jessebraham What the reason CI does not use my `esp-pacs` branch?

**Review by bjoernQ — APPROVED** [review]
LGTM - nice to turn most of that giant macro into normal Rust code

**Review by jessebraham — APPROVED** [review]
Thanks for taking care of these upates!

File: esp-hal/src/dma/gdma.rs [hunk 1]
```diff
@@ -30,541 +30,382 @@ use crate::{
...
+pub struct ChannelTxImpl<const N: u8> {}
 
-                    #[cfg(not(any(esp32c6, esp32h2, esp32s3)))]
-                    let ret = dma.int_raw_ch($num).read().in_done().bit();
-                    #[cfg(any(esp32c6, esp32h2, esp32s3))]
-                    let ret = dma.in_int_raw_ch($num).read().in_done().bit();
+impl<const N: u8> TxChannel<Channel<N>> for ChannelTxImpl<N> {
+    #[cfg(feature = "async")]
+    fn waker() -> &'static embassy_sync::waitqueue::AtomicWaker {
+        static WAKER: embassy_sync::waitqueue::AtomicWaker =
```
Comment (bugadani) [inline esp-hal/src/dma/gdma.rs:341]: Just a post-mortem observation: This change is incorrect, as it only creates a single waker for Tx, and a single one for Rx, regardless of how many DMA channels exist. `static` variables act as though they were defined outside their containing item, so they don't know about any related generic parameters.

**Pull request merged**